src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -282,12 +283,16 @@ def get(self, request: Request, project: Project) -> Response:
                             result["codecovStatusCode"] = 200
                     except requests.exceptions.HTTPError as error:
                         result["codecovStatusCode"] = error.response.status_code
+                        if error.response.status_code == 404:
+                            coverage_expected = False
                         if error.response.status_code != 404:
                             logger.exception(
                                 "Failed to get expected data from Codecov, pending investigation. Continuing execution."
                             )
                     except Exception:
                         logger.exception("Something unexpected happen. Continuing execution.")
+                    # We don't expect coverage data if the integration does not exist (404)
+                    scope.set_tag("codecov.coverage_expected", coverage_expected)
```

Comment:
I would have simply as `codecov.enabled = should_get_codecov_data`. We can always use `codecov.http_code` to determine that there was not coverage for a specific repo.